src/dist/dist.rs

```diff
@@ -644,21 +647,21 @@ fn update_from_dist_<'a>(
         Some(if provided < 1 { 1 } else { provided })
     };
 
-    // We never want to backtrack further back than the nightly that's already installed.
+    // In case there is no allow-downgrade option set
+    // we never want to backtrack further back than the nightly that's already installed.
     //
     // If no nightly is installed, it makes no sense to backtrack beyond the first ever manifest,
     // which is 2014-12-20 according to
     // https://static.rust-lang.org/cargo-dist/index.html.
     //
     // We could arguably use the date of the first rustup release here, but that would break a
     // bunch of the tests, which (inexplicably) use 2015-01-01 as their manifest dates.
-    let first_manifest = old_date
-        .map(|date| {
-            Utc.from_utc_date(
-                &NaiveDate::parse_from_str(date, "%Y-%m-%d").expect("Malformed manifest date"),
-            )
-        })
-        .unwrap_or_else(|| Utc.from_utc_date(&NaiveDate::from_ymd(2014, 12, 20)));
+    let first_manifest = Utc.from_utc_date(&NaiveDate::from_ymd(2014, 12, 20));
+    let old_manifest = if allow_downgrade || old_date == None {
+        first_manifest
+    } else {
+        utc_from_manifest_date(old_date.unwrap())
+    };
```

Comment:
I think this change alters the semantics of a malformed `old_date` from "oh well, use the first manifest" to "panic"

Reply:
Ah right, good catch.

Reply:
I think a malformed `old_date` would have not been catched by the `unwrap_or_else` because the `expect` would have paniced. ;) But I'll fix it